test/functional/collection_tests.js

```diff
@@ -1137,439 +930,334 @@ describe('Collection', function() {
-    },
-
-    // The actual test we wish to run
-    test: function(done) {
-      var self = this;
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
-        poolSize: 1
-      });
-
-      client.connect(function(err, client) {
-        var db = client.db(self.configuration.db);
+      }
+    });
 
-        db.createCollection('shouldCorrectlyReadBackDocumentWithNull', {}, function(
-          err,
-          collection
-        ) {
+    /**
+     * @ignore
+     */
+    it('should correctly read back document with null', {
+      metadata: {
+        requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
+      },
+
+      // The actual test we wish to run
+      test: function(done) {
+        db.createCollection('shouldCorrectlyReadBackDocumentWithNull', {}, (err, collection) => {
           // Insert a document with a date
-          collection.insert({ test: null }, self.configuration.writeConcernMax(), function(err) {
-            test.equal(null, err);
+          collection.insert({ test: null }, this.configuration.writeConcernMax(), err => {
+            expect(err).to.not.exist;
 
-            collection.findOne(function(err) {
-              test.equal(null, err);
+            collection.findOne(err => {
+              expect(err).to.not.exist;
 
               client.close();
               done();
             });
           });
         });
-      });
-    }
-  });
-
-  /**
-   * @ignore
-   */
-  it('should throw error due to illegal update', {
-    metadata: {
-      requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
-    },
-
-    // The actual test we wish to run
-    test: function(done) {
-      var self = this;
-      var client = self.configuration.newClient(self.configuration.writeConcernMax(), {
-        poolSize: 1
-      });
-
-      client.connect(function(err, client) {
-        var db = client.db(self.configuration.db);
+      }
+    });
 
-        db.createCollection('shouldThrowErrorDueToIllegalUpdate', {}, function(err, coll) {
+    /**
+     * @ignore
+     */
+    it('should throw error due to illegal update', {
+      metadata: {
+        requires: { topology: ['single', 'replicaset', 'sharded', 'ssl', 'heap', 'wiredtiger'] }
+      },
+
+      // The actual test we wish to run
+      test: function(done) {
+        db.createCollection('shouldThrowErrorDueToIllegalUpdate', {}, (err, coll) => {
           try {
-            coll.update({}, null, function() {});
+            coll.update({}, null, () => {});
           } catch (err) {
-            test.equal('document must be a valid JavaScript object', err.message);
+            expect('document must be a valid JavaScript object').to.equal(err.message);
           }
 
           try {
-            coll.update(null, null, function() {});
+            coll.update(null, null, () => {});
```

Comment:
When I replaced the two try/catch blocks with `expect(() => coll.update({}, null, () => {})).to.throw('document must be a valid JavaScript object’);` and `expect(() => coll.update(null, null, () => {})).to.throw('selector must be a valid JavaScript object’);`, the test failed with `expected [Function] to throw an error`
I interpreted the try/catch test to mean that there may not be an error, but if it does catch an error it expects a specific error message. However, the expect function seems to expect for an error to exist. Should I modify this new `expect` statement so that it is not always expecting an error, or keep the try/catch?